RxJavaTemplate/config/jacoco.gradle.kts

```
@@ -0,0 +1,115 @@
+apply(plugin = "jacoco")
+
+val fileGenerated = mutableSetOf(
+    "**/R.class",
+    "**/R\$*.class",
+    "**/*\$ViewBinder*.*",
+    "**/*\$InjectAdapter*.*",
+    "**/*Injector*.*",
+    "**/BuildConfig.*",
+    "**/Manifest*.*",
+    "**/*_ViewBinding*.*",
+    "**/*Adapter*.*",
+    "**/*Test*.*",
+    "android/**/*.*"
+)
+
+val packagesExcluded = mutableSetOf(
+    "co/nimblehq/di/**",
+    "co/nimblehq/ui/**/di/**",
+    "com/bumptech/glide"
+)
+
+val fileFilter = fileGenerated + packagesExcluded
+
+val classDirectoriesTree = files(
+    fileTree(project.rootDir) {
+        include(
+            "**/app/build/intermediates/javac/stagingDebug/classes/**",
+            "**/data/build/intermediates/javac/stagingDebug/classes/**",
+            "**/app/build/tmp/kotlin-classes/stagingDebug/**",
+            "**/data/build/tmp/kotlin-classes/stagingDebug/**"
+        )
+        exclude(fileFilter)
+    }
+)
+
+val sourceDirectoriesTree = files(
+    listOf(
+        "${project.rootDir}/app/src/main/java",
+        "${project.rootDir}/data/src/main/java"
+    )
+)
+
+/**
+ * Once enabled [testCoverageEnabled], Jacoco will capture the coverage and store them in
+ * [${project.module}/jacoco.exec]. We need to add all [jacoco.exec] to here.
+ * [${project.module}/build/jacoco/testFlavorDebugTest.exec] won't have the result anymore, so we
+ * can safety get rid of them.
+ * Reference: https://stackoverflow.com/a/67626100/5187859
+ * Issue tracker 1: https://issuetracker.google.com/issues/171125857#comment20
+ * Issue tracker 2: https://issuetracker.google.com/issues/195860510
+ */
+val executionDataTree = fileTree(project.rootDir) {
+    include(
+        "app/jacoco.exec",
+        "common/jacoco.exec"
+    )
+}
+
+tasks.register<JacocoReport>("jacocoTestReport") {
+
+    group = "Reporting"
+    description = "Generate Jacoco coverage reports for Debug build"
+
+    dependsOn(
+        ":app:testStagingDebugUnitTest",
+        ":data:testStagingDebugUnitTest"
+    )
+
+    classDirectories.setFrom(classDirectoriesTree)
+    sourceDirectories.setFrom(sourceDirectoriesTree)
+    executionData.setFrom(executionDataTree)
+
+    reports {
+        xml.isEnabled = true
+        html.isEnabled = true
+        csv.isEnabled = false
+    }
+}
+
+tasks.withType<Test> {
+    configure<JacocoTaskExtension> {
+        isIncludeNoLocationClasses = true
+        /**
```

```suggestion
        /*
```
We use this comment style inside of the block, right? 😄

---

@lydiasama I migrated this comment from @manh-t 's implementation🥲

---

Time to fix that @hoangnguyen92dn 😆